### ISSUE TITLE
[Enhancement]: SageMaker Built In Life Cycle Config 

### ISSUE THREAD

**Issue by oliverguy1** [hashicorp/terraform-provider-aws#39851]
### Description

In the [AWS Create Domain API](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_JupyterLabAppSettings.html#sagemaker-Type-JupyterLabAppSettings-BuiltInLifecycleConfigArn) there is an option to set a "Built In Life Cycle Config" to set the LCC that runs every time a new Jupyter Notebook or Code Editor is set up. 

This does not currently exist within the `aws_sagemaker_domain` resource as of version 5.72.1 of the provider

### Affected Resource(s) and/or Data Source(s)

`aws_sagemaker_domain`

### Potential Terraform Configuration

```terraform
resource "aws_sagemaker_studio_lifecycle_config" "jupyterlab_config" {
  studio_lifecycle_config_name     = "jupyterlab"
  studio_lifecycle_config_app_type = "JupyterLab"
  studio_lifecycle_config_content  = filebase64("${path.module}/jupyterlab.sh")
}

resource "aws_sagemaker_domain" "domain" {
  domain_name = var.sagemaker_domain_name

  default_user_settings {
    code_editor_app_settings {
      built_in_lifecycle_config_arn = aws_sagemaker_studio_lifecycle_config.jupyterlab_config.arn
    }
}
```

### References

[AWS Create Domain API Reference](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_JupyterLabAppSettings.html)
[AWS Create Domain API - BuiltInLifeCycleConfigArn parameter](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_JupyterLabAppSettings.html#sagemaker-Type-JupyterLabAppSettings-BuiltInLifecycleConfigArn)

### Would you like to implement a fix?

No

**Comment by deepakbshetty** [comment]
Hi @oliverguy1

I believe this is now released in 5.73.0 [ https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.73.0 ]

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sagemaker_domain#built_in_lifecycle_config_arn

**Comment by justinretzolk** [comment]
Hey @oliverguy1 👋 Thank you for taking the time to raise this! This was indeed [released in `5.73.0`](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5730-october-24-2024) via #39774. With that in mind, I'm going to close this issue. If you have any future enhancement requests, please do let us know!
